web/api/report_server.thrift

```diff
@@ -203,15 +203,17 @@ typedef list<RunHistoryData> RunHistoryDataList
  * If exactMatch field is True it will use exact match for run names.
  */
 struct RunHistoryFilter {
-  1: list<string> tagNames,   // Part of the tag names.
+  1: list<string> tagNames,     // Part of the tag names.
+  2: optional list<i64> tagIds, // Tag ids.
```

Comment:
what is the recommended usage here? Both values can be set by the client (partial tag name and tagid)? The union of the matched history elements are returned?

Reply:
`tagNames` can be used to search for a run history entry by tag name (on the UI there is a search option which do this).

In this patch tag ids will be placed in the url. With the `tagIds` you can get run history entries for the given tags.

When you use both it will work the same way as other filters: it will filter by tagNames AND tagIds.